consensus/istanbul/validator/weighted.go

```diff
@@ -553,6 +572,36 @@ func (valSet *weightedCouncil) F() int {
 
 func (valSet *weightedCouncil) Policy() istanbul.ProposerPolicy { return valSet.policy }
 
+func (valSet *weightedCouncil) Update(hash common.Hash, blockNum uint64) error {
```

Comment:
Please add some description about this function.

Reply:
I will add some comments.